test/e2e/suite_test.go

```diff
@@ -61,45 +98,174 @@ var _ = BeforeSuite(func(ctx SpecContext) {
 	err = v1alpha1.AddToScheme(scheme.Scheme)
 	Expect(err).NotTo(HaveOccurred())
 
-	cfg, err := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(
-		clientcmd.NewDefaultClientConfigLoadingRules(),
-		&clientcmd.ConfigOverrides{},
-	).ClientConfig()
-	Expect(err).NotTo(HaveOccurred())
-
-	k8sClient, err = client.New(cfg, client.Options{Scheme: scheme.Scheme})
-	Expect(err).NotTo(HaveOccurred())
-	Expect(k8sClient).NotTo(BeNil())
-
-	testManagedZone = &v1alpha1.ManagedZone{}
-	err = k8sClient.Get(ctx, client.ObjectKey{Namespace: testNamespace, Name: testManagedZoneName}, testManagedZone)
-	Expect(err).NotTo(HaveOccurred())
+	loadClusters(ctx)
+	Expect(testDNSProvider).NotTo(BeEmpty())
+	Expect(testZoneDomainName).NotTo(BeEmpty())
+	Expect(testClusters).NotTo(BeEmpty())
+	for i := range testClusters {
+		Expect(testClusters[i].testManagedZones).NotTo(BeEmpty())
+	}
 
 	testSuiteID = "dns-op-e2e-" + GenerateName()
 
 	geoCode := "EU"
-	if testDNSProvider == "gcp" {
+	if testDNSProvider == "google" {
 		geoCode = "europe-west1"
 	}
 	SetTestEnv("testGeoCode", geoCode)
 })
 
+// setConfigFromEnvVars loads test suite runtime configurations from env vars.
+// dnsManagedZoneName managed zone name expected to exist in each test namespace (i.e. dev-mz-aws).
+// dnsNamespaces test namespaces, comma seperated list (i.e. dns-operator-1,dns-operator-2)
+// deploymentCount number of test namespaces expected. Appends an index suffix to the dnsNamespaces, only used if dnsNamespaces is a single length array.
+//
+// Examples:
+// dnsNamespaces=dns-operator deploymentCount=<unset> = dnsNamespaces=dns-operator
```

Comment:
I'm struggling with the syntax here, why is there a floating `=`?

Reply:
I had a go at trying to make the example a bit clearer. They are trying to show if you give these inputs, this is what the resulting config will be.